### PR TITLE
PCHR-3426: Fix buttons borders

### PR DESCRIPTION
**Pull request by igorpavlov-zz**
## Overview

This PR fixes the styles for buttons inside `btn-group`. This PR is a part of https://github.com/civicrm/civihr/pull/2536.

## Before

Note: the zoom on the screenshot is 2x for convenience.

The border on the blue button is due to the ".btn-group" container.

image

## After 

The border on the blue button has gone.

image

## Technical Details

There are a lot of gaps of 8px in the mockups, so having that variable is handy.

```css
$crm-gap-medium: 8px !default;
```

### Button groups styles enhancement

A style was moved to a partial related to buttons instead of button groups:

```css
.btn .caret { // this is not about button groups, but buttons
  margin: 0 5px;
```

Remove border from buttons:

```
.btn {
  box-shadow: none;
  border-color: $input-border; // remove
```

Add styles for active labels:

```css
.active {
  background-color: $gray;
  border-color: transparent;
  color: $crm-white;
```

And finally, discard the default Bootstrap rule for border for all buttons except the ones that are outlined:

```css
button:not([class*='-outline']) {
  border-color: transparent;
```

### Add a gap variable

As per mockups, the 8px gap is reused in several places:

image

1) Around record buttons 
2) Between record buttons

image

1) Between circles

So a variable has been added.

```css
$crm-gap-medium: 8px !default;
```

### Remove buttons glowing on focus

Before:
image

After: 

image

```css
.btn {
  // ...

  &:focus, // add that
  &:active,
  &.active {
    @include box-shadow(none);
    outline: none; // and that
```

## Comments

This PR also disables the rule to force exactly one space after : in SASS and updates 'stylelint-scss' to 2.5.0 version.

---- 

✅Backstop tests - passed
✅CSS distributive files - included

### CODE REVIEW

**Review by AkA84 — COMMENTED** [review]
Sorry @igorpavlov but the PR description needs a pretty serious overhaul

---

Starting from the absolute basics: there are no screenshots attached to the PR. You are fixing and adding styles, but yet you provide no screenshots to show what you are talking about

---

You are not providing any overview of your PR in the Overview section, instead you go straight into technical details, which should be instead in a Technical Details section. In the Overview you should tell me in non-technical details what this PR was for (and maybe why), and show me any visual changes that you introduced (= screenshots). I should be able to read the Overview and know exactly what you did and why without having to bother to read technical stuff. 

---

You mention 

> There are a lot of gaps of 8px in the mockups,

can you show what you are talking about? What are these gaps? Where? Any PR reviewer would have no idea if you don't show them

---

Finally, the section in which you explain the technical changes to the button groups styles is verging on being unintelligible at times. It's extremely hard to follow given that you have this weird mix of source code and documentation all bundled together which doesn't work at all.

Again, I want to stress that the "technical details" section is not a place to dump entire chunks of source code with some annotations put on top of it. Nor it requires to document every line of the code.

If you think that a part of your solution is worth being documented, then please write a proper, well structured documentation in a way that you yourself would want to read. Something like

## Technical details
### Enhanced button group details
Some overview, if necessary
#### First change
Lorem ipsum dolor sit amet
```scss
.btn-group {
  // ..
  some-property: some-value;
}
```
(screenshot to show the outcome)

#### Second change
_(etc)_

It doesn't have to be structured exactly like that, but you can see that the point is that the idea is that there is a brief explanation of the change, then the code itself (annotated, if necessary), and then a screenshot if we're talking about a visual change

please see https://github.com/compucorp/org.civicrm.shoreditch/pull/85 as an example

File: scss/bootstrap/overrides/style/_button-groups.scss [hunk 1]
```diff
@@ -1,27 +1,16 @@
...
-  margin: 0 5px;
-}
-
 .btn-group {
-
-  .btn {
-    box-shadow: none;
+  label,
+  label:hover {
     border-color: $input-border;
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_button-groups.scss]: i don't get this, why are now `<label>` elements getting a border color, but only if they are in a `.btn-group` element? When is that going to happen? Are we talking about something like
```html
<div class="btn-group">
  <label>some label</label>
</div>
```
?

File: scss/bootstrap/overrides/style/_button-groups.scss [hunk 2]
```diff
@@ -1,27 +1,16 @@
...
-      }
-    }
-
-    // Adds missing border color on button hover:
-    &:hover {
-      border-color: $crm-grayblue-darker;
-    }
+  .active {
+    background-color: $gray;
+    border-color: transparent;
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_button-groups.scss]: Is this rule necessary? I see it's being overridden by
```scss
#bootstrap-theme .btn.active {
  border-color: transparent;
}
```
and
```scss
#bootstrap-theme .btn-group button:not([class*='-outline']) {
  border-color: transparent;
}
```

File: scss/bootstrap/overrides/style/_button-groups.scss [hunk 3]
```diff
@@ -1,27 +1,16 @@
...
+    color: $crm-white;
+  }
 
-    &.active {
-      background-color: $gray;
-      color: $crm-white;
-    }
+  button:not([class*='-outline']) {
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_button-groups.scss]: Is this style necessary? I see it's being overridden by
```scss
#bootstrap-theme .btn.active {
  border-color: transparent;
}
```

File: scss/bootstrap/overrides/style/_buttons.scss [hunk 4]
```diff
@@ -90,12 +81,14 @@
 }
 
 .btn-danger {
-
   &:hover {
     background-color: darken($btn-danger-bg, $crm-darken-percentage);
   }
 }
 
+.btn .caret {
```
Comment (AkA84) [inline scss/bootstrap/overrides/style/_buttons.scss:89]: what is this `.caret` class coming from? Is it from bootstrap?

**Review by AkA84 — CHANGES_REQUESTED** [review]

File: scss/bootstrap/overrides/style/_button-groups.scss (hunk 1 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_button-groups.scss]: Yes, this is a current markup of leave request modal.

https://github.com/civicrm/civihr/blame/staging/uk.co.compucorp.civicrm.hrleaveandabsences/views/shared/components/leave-request-popup/details-tab/common.html#L7

I can refactor it to make it work like normal button group.

File: scss/bootstrap/overrides/style/_button-groups.scss (hunk 2 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_button-groups.scss]: Correct, this style is not needed. Removed.

File: scss/bootstrap/overrides/style/_button-groups.scss (hunk 1 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_button-groups.scss]: I have refactored the Multiple Day Selector in the modal https://github.com/civicrm/civihr/pull/2536 so now if we use a standard Bootstrap markup, we don't need these extra styling.

File: scss/bootstrap/overrides/style/_button-groups.scss (hunk 3 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_button-groups.scss]: Yes, removed.

File: scss/bootstrap/overrides/style/_buttons.scss (hunk 4 above)
Comment (igorpavlov-zz) [reply scss/bootstrap/overrides/style/_buttons.scss:89]: Yes: https://getbootstrap.com/docs/3.3/components/#dropdowns (one of uses)

**Review by AkA84 — APPROVED** [review]
